drivers/ethernet/Kconfig

```diff
@@ -5,12 +5,20 @@
 
 menuconfig ETH_DRIVER
 	bool "Ethernet drivers"
-	default y
-	depends on NET_L2_ETHERNET
-	depends on !NET_TEST
```

Comment:
The purpose of the `NET_TEST` is to prevent Ethernet drivers to be taken into use when building network tests, as those use simulated Ethernet drivers and are self contained. Using real Ethernet driver will just cause problems in the tests if there are extra network interfaces created. I don't think we should remove `NET_TEST` here.